### DISCUSSION TITLE
Query affected 2 rows while 1 expected! This might mean that your query lacks a proper WHERE statement or a TRIGGER changes number of affected rows. In the latter case, you may try adding "SET NOCOUNT ON" to your trigger code

### DISCUSSION THREAD

**Discussion by minhhungit** [serenity-is/Serenity#6342] [Tricks and Samples]
My table is using a sql trigger to update a computed column, and when I update the update on entity dialog, it threw this error

```text
Query affected 2 rows while 1 expected! 
This might mean that your query lacks a proper WHERE statement or a TRIGGER changes number of affected rows. 
In the latter case, you may try adding "SET NOCOUNT ON" to your trigger code
```

After checking serenity source code I see that it is because serenity has mechanism to verify saving by using `ExpectedRows` at [here](https://github.com/serenity-is/Serenity/blob/f1ed76106c7443fc3da7c10e4606bdfb99666324/src/Serenity.Net.Services/RequestHandlers/Save/SaveRequestHandler.cs#L91) 

I am not sure what is correct way to overwrite ExpectedRows, so I tried to modify `ExecuteSave` like bellow, this might help some one.
The idea is copy code from serenity source code and change ExpectedRows. Make sure that you copy code from corresponding serenity version you are using for your project

Here is the code for serenity .net framework, version 3.13.7.0

```csharp
private class MySaveHandler : SaveRequestHandler<MyRow>
{
    protected override void ExecuteSave()
    {
        if (IsUpdate)
        {
            var field = (Field)((Row as IIdRow).IdField);
            if (field.IndexCompare(Old, Row) != 0)
            {
                SqlUpdate sqlUpdate = new SqlUpdate(Row.Table);
                sqlUpdate.Set(Row);
                sqlUpdate.Where(field == new ValueCriteria(field.AsSqlValue(Old)));
                sqlUpdate.Execute(Connection, ExpectedRows.Ignore);
            }
            else
            {
                Connection.UpdateById(Row, ExpectedRows.Ignore);
            }

            Response.EntityId = field.AsObject(Row);
            InvalidateCacheOnCommit();
        }
        else
        {
            base.ExecuteSave();
        }        
    }
}
```

i want to share my experience but If someone knows a way to custom ExpectedRows without changing code like above please share me, thank you

**Comment by VictorTomaili** [comment]
The exception says `You may try adding "SET NOCOUNT ON" to your trigger code`

**Reply by minhhungit** [reply]
oh my bad, I tried again, this times it works, **thank you**
I think before it did not work because I put "SET NOCOUNT ON" wrong place, instead **inside** the trigger, I put it on **top** of the trigger.

![image](https://user-images.githubusercontent.com/2279508/161042651-08da36d4-ddf3-40e0-ac12-e4426bcb719b.png)
